ckanext/s3filestore/uploader.py

```diff
@@ -628,15 +627,21 @@ def update_visibility(self, id, target_acl=None):
             return
 
         for upload in resource_objects['Contents']:
-            is_public_read = self.is_key_public(upload['Key'])
+            upload_key = upload['Key']
+            log.debug("Setting visibility for key [%s], current object is [%s]", upload_key, current_key)
+            if upload_key == current_key:
```

Comment:
Can we put this behind a config flag
```ckanext.s3filestore.hideOldResourceObject = true/false #default value: true```
if true if not same as db key then set private
if false, same as original functionality (all is same as active file)

another option for a 'cleaned up' s3 is

```ckanext.s3filestore.deleteNonLinkedResource = true/false #default value: false```
```ckanext.s3filestore.deleteNonLinkedResourceAge = <number of days old> #default value: 90```

if a file is found that does not match database and it is older than deleteNonLinkedResourceAge, then delete file so it is treated like it was overwritten with possible backup window.

for our purposes we would have it set to true. As we use s3 versioning we have 90+ days to recover the file if requested. 

This keeps our blob storage size down and only list what is still active.

Reply:
seems you can't 'modify' objects to have a marker, so deleteNonLinkedResourceAge can't be done if true it will purge dead data.

Reply:
I've added a config value to control the ACLs of non-current files. Default is 'private'.

I'm unclear, though, on why deleting old objects would require modifying tags. Couldn't we just rely on the built-in "Last Modified" metadata?

Reply:
ok, thats true, if its fresh it is kept for x days and then deleted which will give it more grace period.